k8s/migration/internal/controller/rdmdisk_controller.go

```diff
@@ -0,0 +1,347 @@
+				log.Error(err, "unable to update RDMDisk status")
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{}, nil
+		}
+
+		// Validation passed, move to Pending phase
+		rdmDisk.Status.Phase = "Pending"
+		meta.SetStatusCondition(&rdmDisk.Status.Conditions, metav1.Condition{
+			Type:    "Validated",
+			Status:  metav1.ConditionTrue,
+			Reason:  "ValidationPassed",
+			Message: "All required fields validated",
+		})
+		if err := r.Status().Update(ctx, rdmDisk); err != nil {
+			log.Error(err, "unable to update RDMDisk status")
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{Requeue: true}, nil
+
+	case "Pending":
+		// Check if ImportToCinder is true, move to Managing phase
+		if rdmDisk.Spec.ImportToCinder {
+			rdmDisk.Status.Phase = "Managing"
+			meta.SetStatusCondition(&rdmDisk.Status.Conditions, metav1.Condition{
+				Type:    "MigrationStarted",
+				Status:  metav1.ConditionTrue,
+				Reason:  "ImportToCinderEnabled",
+				Message: "Starting migration to Cinder Importing LUN",
+			})
+			if err := r.Status().Update(ctx, rdmDisk); err != nil {
+				log.Error(err, "unable to update RDMDisk status")
+				return ctrl.Result{}, err
+			}
+			return ctrl.Result{Requeue: true}, nil
+		}
+		return ctrl.Result{}, nil
+
+	case "Managing":
+		ctxlog := log.WithName(constants.RDMDiskControllerName)
+
+		if rdmDisk.Spec.ImportToCinder && rdmDisk.Status.CinderVolumeID == "" {
+			// Create the RDM disk object with required fields
+			rdmDiskObj := vm.RDMDisk{
+				DiskName:          rdmDisk.Name,
+				VolumeRef:         rdmDisk.Spec.OpenstackVolumeRef.Source,
+				CinderBackendPool: rdmDisk.Spec.OpenstackVolumeRef.CinderBackendPool,
+				VolumeType:        rdmDisk.Spec.OpenstackVolumeRef.VolumeType,
+			}
+			openstackcreds := &vjailbreakv1alpha1.OpenstackCreds{}
+			openstackCredsName := client.ObjectKey{
+				Namespace: req.Namespace,
+				Name:      rdmDisk.Spec.OpenstackVolumeRef.OpenstackCreds,
+			}
+			if err := r.Get(ctx, openstackCredsName, openstackcreds); err != nil {
+				if apierrors.IsNotFound(err) {
+					ctxlog.Info("Resource not found, likely deleted", "openstackcreds", openstackCredsName)
+					return ctrl.Result{}, nil
+				}
+				ctxlog.Error(err, "Failed to get OpenstackCreds resource", "openstackcreds", openstackCredsName)
+				return ctrl.Result{}, err
+			}
+			ctxlog.V(1).Info("Retrieved OpenstackCreds resource", "openstackcreds", openstackCredsName, "resourceVersion", openstackcreds.ResourceVersion)
+			openstackClient, err := utils.GetOpenStackClients(ctx, r.Client, openstackcreds)
+			if err != nil {
+				return ctrl.Result{}, handleError(ctx, r.Client, rdmDisk, "Error", "OpenStackClientCreationFailed", "Failed to create OpenStack client from options", err)
+			}
+			osclient := migrateutils.OpenStackClients{
+				BlockStorageClient: openstackClient.BlockStorageClient,
+				ComputeClient:      openstackClient.ComputeClient,
+				NetworkingClient:   openstackClient.NetworkingClient,
+			}
```

Comment:
The `utils.GetOpenStackClients` returns `utils.OpenStackClients`. We should use that directly instead of converting it to migrateutils.OpenStackClients. 

Both structs are same so we should rather modify the ImportLUNToCinder to use `utils.OpenStackClients`

Reply:
WaitForVolume is in migrateutils of v2v-helper , hence the package is used , so that WaitforVolume Func to be reused